samples/nrf9160/modem_shell/src/main.c

```diff
@@ -87,6 +90,17 @@ static void mosh_print_version_info(void)
 
 static void button_handler(uint32_t button_states, uint32_t has_changed)
 {
+	if (has_changed & button_states & DK_BTN1_MSK) {
+		shell_print(shell_global, "Button 1 pressed - raising a kill signal");
+		k_poll_signal_raise(&mosh_signal, MOSH_SIGNAL_KILL);
+#if defined(CONFIG_MOSH_WORKER_THREADS)
+		th_ctrl_kill_em_all();
+#endif
+	} else if (has_changed & ~button_states & DK_BTN1_MSK) {
+		shell_print(shell_global, "Button 1 released - resetting a kill signal");
+		k_poll_signal_reset(&mosh_signal);
+	}
```

Comment:
There may be a good reason for doing this, but just wondering why signal is raised when button is pressed and reset when the button is released? Why not just raise the signal when button is pressed and that's it? I guess it would be left raised if nothing would be running at that moment, but the signal could be reset when a new operation is started.